motionplan/tpSpaceRRT.go

```diff
@@ -450,20 +455,22 @@ func (mp *tpSpaceRRTMotionPlanner) getExtensionCandidate(
 	// Get the distance function that will find the nearest RRT map node in TP-space of *this* PTG
 	ptgDistOpt := mp.algOpts.distOptions[curPtg]
 
-	if nearest == nil {
+	if nearestNeighborWithSolution == nil {
 		// Get nearest neighbor to rand config in tree using this PTG
 		// TODO: running nearestNeighbor actually involves a ptg.Solve() call, duplicating work.
-		nearest = nm.nearestNeighbor(ctx, ptgDistOpt, randPosNode, rrt)
-		if nearest == nil {
+		// TODO: COME BACK HERE
+		nearestNeighborWithSolution = nm.nearestNeighborWithDistanceAndSolution(ctx, ptgDistOpt, randPosNode, rrt)
+		if nearestNeighborWithSolution.node == nil {
 			return nil, errNoNeighbors
 		}
 	}
+	nearestNode := nearestNeighborWithSolution.node
 	// TODO: We could potentially improve solving by first getting the rough distance to the randPosNode to any point in the rrt tree,
 	// then dynamically expanding or contracting the limits of IK to be some fraction of that distance.
 
 	// Get cartesian distance from NN to rand
 	var targetFunc ik.StateMetric
-	relPose := spatialmath.PoseBetween(nearest.Pose(), randPosNode.Pose())
+	relPose := spatialmath.PoseBetween(nearestNode.Pose(), randPosNode.Pose())
 	targetFunc = mp.algOpts.goalMetricConstructor(relPose)
 	seedDist := relPose.Point().Norm()
 	seed := tpspace.PTGIKSeed(curPtg)
```

Comment:
All of this on down to line 492 should be able to be replaced with `bestNode := nearestNeighborWithSolution.solution` right?